Automattic/newspack-ads
# Add ability to exclude posts and pages from ads

Closes https://github.com/Automattic/newspack-issues/issues/115. 

**This PR should be tested together with https://github.com/Automattic/newspack-plugin/pull/482.**

This PR adds a toggle for posts and pages that suppresses ads on the post/page. Currently it's only hooked up to the Newspack Google Ads integration, but we should be able to hook it up to the other supported ad providers if needed.

#### To test:

1. Set up some ads that will display on a post or page: at least one global ad, a widget ad, and an ad block.
<img width="1171" alt="Screen Shot 2020-03-11 at 10 54 42 AM" src="https://user-images.githubusercontent.com/7317227/76452373-4574a680-638e-11ea-8efd-5065581c913a.png">

2. In the Document panel of the post, toggle ads off:
<img width="272" alt="Screen Shot 2020-03-11 at 10 54 53 AM" src="https://user-images.githubusercontent.com/7317227/76452417-558c8600-638e-11ea-9191-7c1c1d3af7a1.png">

3. Visit post/page again. Observe no ads:
<img width="1142" alt="Screen Shot 2020-03-11 at 10 55 08 AM" src="https://user-images.githubusercontent.com/7317227/76452450-63420b80-638e-11ea-8809-571dbcb194c1.png">
